From dd0b4441abb6c80797d39326381d173671ddf7ac Mon Sep 17 00:00:00 2001
From: mj <mj@maxlinear.com>
Date: Thu, 4 Sep 2025 05:06:25 -0700
Subject: [PATCH] WLANRTSYS-92970 compare lower freq for channel overlap
 validation

Issue : validated primary channel frequency for channel overlap which is
not valid. Some channels are skipped in channel overlap.

Fix: Validate Lower channel frequency in channel overlap
---
 src/ap/mxl_acs_proprietary.c | 17 ++---------------
 1 file changed, 2 insertions(+), 15 deletions(-)

diff --git a/src/ap/mxl_acs_proprietary.c b/src/ap/mxl_acs_proprietary.c
index aaa8201ed..d76860f4a 100644
--- a/src/ap/mxl_acs_proprietary.c
+++ b/src/ap/mxl_acs_proprietary.c
@@ -4381,8 +4381,6 @@ acs_get_failsafe_indexes_per_bw(struct hostapd_iface *iface,
 	int i;
 	int freq;
 	int sec_best_freq;
-	int op_class;
-	int sec_op_class;
 	struct acs_candidate_table *candidate = iface->mxl_iface.candidates;
 	int best20 = INT_MAX, best40 = INT_MAX, best80 = INT_MAX, best160 = INT_MAX;
 	int best20_util = 0, best40_util = 0, best80_util = 0, best160_util = 0;
@@ -4397,19 +4395,8 @@ acs_get_failsafe_indexes_per_bw(struct hostapd_iface *iface,
 			continue;
 
 		if (iface->mxl_iface.sec_best_ch.first_rank_idx && iface->mxl_iface.sec_best_ch.flag) {
-			hostapd_get_op_class(candidate[i].primary, candidate[i].width, &op_class);
-			freq = ieee80211_chan_to_freq(NULL, op_class, candidate[i].primary);
-			hostapd_get_op_class(candidate[iface->mxl_iface.sec_best_ch.first_rank_idx].primary, candidate[iface->mxl_iface.sec_best_ch.first_rank_idx].width, &sec_op_class);
-			sec_best_freq = ieee80211_chan_to_freq(NULL, sec_op_class, candidate[iface->mxl_iface.sec_best_ch.first_rank_idx].primary);
-
-			if (freq < 0) {
-				wpa_printf(MSG_ERROR,"Could not convert op_class %d channel %d to operating frequency",
-							  op_class, candidate[i].chan);
-			}
-			if (sec_best_freq < 0) {
-				wpa_printf(MSG_ERROR,"Could not convert op_class %d second best channel %d to operating frequency",
-							  sec_op_class, candidate[iface->mxl_iface.sec_best_ch.first_rank_idx].primary);
-			}
+			freq = candidate[i].freq;
+			sec_best_freq = candidate[iface->mxl_iface.sec_best_ch.first_rank_idx].freq;
 			if (channels_overlap(sec_best_freq, candidate[iface->mxl_iface.sec_best_ch.first_rank_idx].width, freq, candidate[i].width)) {
 				wpa_printf(MSG_DEBUG, "channel overlapped i = %d  first freq %d, width : %d : candidates[i].freq %d, width : %d", i, sec_best_freq, candidate[iface->mxl_iface.sec_best_ch.first_rank_idx].width, freq, candidate[i].width);
 				continue;
-- 
2.43.0

